src/module.cpp

```diff
@@ -128,3 +129,33 @@ void Module::VerifyFunction() {
 		}
 	}
 }
+
+const std::string Module::MapSpecie(const std::pair<std::string, int> &specie) {
+	std::string output;
+	if (name != mainModuleName) {
+		if (specie.second != 1) {
+
+			output +=
+					std::to_string(specie.second) + name + "_" + specie.first + " + ";
+		} else {
+			output += name + "_" + specie.first + " + ";
+		}
+	} else {
+		if (specie.second != 1) {
+			output += std::to_string(specie.second) + specie.first + " + ";
+		} else {
+			output += specie.first + " + ";
+		}
+	}
+	return output;
+}
+const std::string Module::MapConcs(const std::pair<std::string, int> &concs) {
+	std::string output;
+	if (name != mainModuleName) {
```

Comment:
Again a case of why you should split up the previous function. If you had a function, which only had the function of mapping a specie name, you could have used it here, instead of repeating the `if name != mainModuleName)` logic.